src/idl_gen_cpp.cpp

```diff
@@ -1650,7 +1650,7 @@ class CppGenerator : public BaseGenerator {
       const auto &type = field.value.type;
       code_.SetValue("PARAM_VALUE", GetDefaultScalarValue(field, false));
       if (field.IsScalarOptional())
-        code_.SetValue("PARAM_TYPE", GenOptionalDecl(type));
+        code_.SetValue("PARAM_TYPE", GenOptionalDecl(type) + " ");
```

Comment:
I think it should be fixed in line 1657.

Reply:
I think that'll cause vectors to look like `std::vector<Foo> * foo = nullptr`, because this function considers the spacing between the type and parameter name to be part of the type to make pointers look right. I also tried fixing it in GenOptionalDecl but other functions didn't make the same assumption.

Reply:
Agree, thanks.